src/java/org/apache/cassandra/db/marshal/UserType.java

```diff
@@ -67,7 +68,10 @@ public UserType(String keyspace, ByteBuffer name, List<FieldIdentifier> fieldNam
         {
             String stringFieldName = fieldNames.get(i).toString();
             stringFieldNames.add(stringFieldName);
-            fieldSerializers.put(stringFieldName, fieldTypes.get(i).getSerializer());
+            TypeSerializer<?> previous = fieldSerializers.put(stringFieldName, fieldTypes.get(i).getSerializer());
+            assert previous == null : String.format("Duplicate names found for UDT(%s.%s) and column %s",
```

Comment:
super nit/alternative name: `existing`

Reply:
Do we use the `UDT(keyspace, type name)` format anywhere else? Maybe it would be clear enough to just get rid of the parens and have it read "...found for UDT ks.name..."

Reply:
If a user happened to have created an invalid schema in the field, would this prevent startup?

Reply:
> Do we use the UDT(keyspace, type name) format anywhere else?

no, we do `"keyspace"."name"`, can remove the params

> If a user happened to have created an invalid schema in the field, would this prevent startup?

Yes, it would.  If there is a conflict then you would get serialization issues reading right now or get incorrect results.

Reply:
How would the user recover from that?

Reply:
should I add the same "behavior" stuff I add to everything else?

```
enum ConflictBehavior
{
  LOG {
  public void onConflict(String ks, String name, String column, TypeSerializer<?> previous, TypeSerializer<?> latest) {
  logger.error("life isn't fair... your types are messed up yo");
}
},
  REJECT {
  public void onConflict(String ks, String name, String column, TypeSerializer<?> previous, TypeSerializer<?> latest) {
throw all the things
}
};

  public abstract void onConflict(String ks, String name, String column, TypeSerializer<?> previous, TypeSerializer<?> latest);
}
```

Reply:
added the enum, we default to reject but have the following error

```
 java.lang.AssertionError: Duplicate names found in UDT a.b for column a; to resolve set -Dcassandra.type.udt.conflict_behavior=LOG on startup and remove the type
```

So should say how to fix.